spring-transaction/src/main/kotlin/org/jetbrains/exposed/spring/SpringTransactionManager.kt

```diff
@@ -115,6 +130,50 @@ class SpringTransactionManager(
                 addLogger(StdOutSqlLogger)
             }
         }
+
+        lateinit var con: Connection
+        try {
+            if (!trxObject.hasConnectionHolder() ||
+                trxObject.getConnectionHolder().isSynchronizedWithTransaction
+            ) {
+                trxObject.setConnectionHolder(
+                    ConnectionHolder(ExposedConnectionHandle(transaction))
+                )
```

Comment:
This `ExposedConnectionHandle` is the thing that makes this all work. See comment downstairs.